testctrl/proto/test_sessions.proto

```diff
@@ -0,0 +1,193 @@
+  // specify "2" replicas. Two identical versions will be available to the test.
+  //
+  // This field is ignored for driver components. It may eventually be supported
+  // on server components, but this is not yet available.
+  int32 replicas = 3;
+}
+
+message StartTestSessionRequest {
+  // The test scenario to run as a part of the test session.
+  grpc.testing.Scenario scenario = 1;
+
+  // The component which orchestrates running the test scenarios amongst
+  // workers.
+  Component driver = 2;
+
+  // The components that are required as part of the test. Normally, this will
+  // involve at least one server and a number of clients.
+  repeated Component workers = 3;
+
+  // An optional location where the result should be written.
+  oneof result_destination {
+    // The name of the BigQuery table where result should be written.
+    string bq_results_table = 4;
+
+    // The fully qualified GCS URI where a result file should be written.
+    string gcs_results_file_uri = 5;
+  }
+}
+
+message TestSessionResult {
+  // The result for the scenario.
+  grpc.testing.ScenarioResult scenario_result = 1;
+
+  // The amount of time that this session lived, including all actions from
+  // scheduling to termination.
+  google.protobuf.Duration time_elapsed = 2;
+}
+
+message Event {
+  enum Kind {
+    // An infrastructure problem. This is outside of the test runner's control,
+    // and a bug should be filed if it is encountered.
+    INTERNAL_ERROR = 0;
+
+    // Waiting for an executor to process the test session.
+    QUEUE = 1;
+
+    // An executor has been assigned to provision and monitor the session;
+    // however, work has not yet begun on this event's subject.
+    ACCEPT = 2;
+
+    // Reserving and configuring resources for the event's subject.
+    PROVISION = 3;
+
+    // Resources are responding with a healthy signal. However, this is not
+    // indicative of a running test.
+    RUN = 4;
+
+    // Resources terminated or released as expected. It does not indicate that
+    // the test was successful or the result was recorded.
+    DONE = 5;
+
+    // An irrecoverable error has caused the subject of the event to terminate.
+    ERROR = 6;
+  }
+
+  // The name of the subject of the event. This may be the name of a test
+  // session or the name of one of its components.
+  string subject = 1;
+
+  // The kind of event.
+  Kind kind = 2;
+
+  // The point in time when the event was noticed.
+  google.protobuf.Timestamp time = 3;
+}
+
+message TestSessionMetadata {
+  // The latest event for the test session as a whole, assuming it has not been
+  // terminated.
+  Event latest_event = 1;
+
+  // The component error event that caused the session to error, if applicable.
+  Event component_error_event = 2;
+
+  // A URL with a file that contains all events for all components in the
+  // session.
+  string event_log_url = 3;
+
+  // The name of the kubernetes cluster where tests were run.
```

Comment:
It looks like gke_cluster may not be needed, I would leave it out here.  And probably the same for event_log_url.